### DISCUSSION TITLE
Is sim efficiency dependent on CPU type

### DISCUSSION THREAD

**Discussion by dheligman** [NanoComp/meep#2033] [Q&A]
Hi all,

I was wondering if there is a efficiency dependence on CPU type. The current workstation I am using for my simulations use AMD cores as opposed to intel. Would this affect the speed of the simulations?

Best,
Daniel

**Comment by oskooi** [comment]
For running Meep simulations on a single, multi-core, shared-memory machine, the impact of the hardware on the simulation runtime can be critical in particular the size of the CPU cache (L1/L2/L3) as well as the size and speed of the memory. This is because Meep simulations are primarily based on reading and writing to C/C++ arrays (of the fields and materials during time stepping) which means they are generally limited by [memory bandwidth](https://en.wikipedia.org/wiki/Memory_bandwidth). 

Note that Meep provides a number of built-in features for performance optimization including: (1) support for [single-precision floating point for the fields and materials arrays](https://meep.readthedocs.io/en/latest/Build_From_Source/#floating-point-precision-of-the-fields-and-materials-arrays) (a compile-time option) which can typically provide a factor of two speedup in the time-stepping rate compared to the default double-precision floating point, (2) [loop nest optimization](https://en.wikipedia.org/wiki/Loop_nest_optimization) for the fields updates (step curls: D/B from H/E, and constitutive: E/H from D/B) via the [`loop_tile_base_db` and `loop_tile_base_eh` parameters of the `Simulation` object constructor](https://meep.readthedocs.io/en/latest/Python_User_Interface/#the-simulation-class), and (3) decimation of the DTFT field updates via the `decimation_factor` parameter of the DFT routines such as `add_flux`, `add_mode_monitor`, `add_dft_fields`, etc. (As an additional enhancement, Meep supports [multithreading via OpenMP](https://meep.readthedocs.io/en/latest/Parallel_Meep/#different-forms-of-parallelization) of the fields updates during time stepping.)

**Reply by stevengj** [reply]
In other words, yes it will make a difference, but there's no reason to specifically prefer Intel over AMD (or vice versa) that I know of.  You just want the fastest machine with the most memory bandwidth that you can get your hands on.
